### PR TITLE
[ACS] Fix the SPN Insufficient Permission Issue for Virtual Kubelet installed by `aks install-connector`

### PR DESCRIPTION
**Pull request by robbiezhang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5583
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-acs/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+2.0.27
+++++++
+* use the virtual-kubelet-for-aks helm chart for `aks install-connector` by default, and fix the service principal insufficient permission to create ACI container group issue
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/HISTORY.rst]: Could you separate this into two bullet points?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py [hunk 2]
```diff
@@ -235,33 +235,45 @@
         - name: --service-principal
           type: string
           short-summary: Service principal used for authentication to Azure APIs.
-          long-summary:  If not specified, a new service principal is created and cached at
-                         {sp_cache} to be used by subsequent `az aks` commands.
+          long-summary:  If not specified, use the AKS service principal defined in the file
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_help.py]: Typo: virtual

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -313,62 +313,75 @@ def k8s_install_cli(cmd, client_version='latest', install_location=None):
...
     logger.warning("Deploying the ACI connector for '%s' using Helm", os_type)
     try:
-        subprocess.call(["helm", "install", url_chart, "--name", helm_release_name, "--set", "env.azureClientId=" +
-                         service_principal + ",env.azureClientKey=" + client_secret + ",env.azureSubscriptionId=" +
-                         subscription_id + ",env.azureTenantId=" + tenant_id + ",env.aciResourceGroup=" +
-                         aci_resource_group + ",env.aciRegion=" + aci_region + ",image.tag=" + image_tag +
-                         ",env.nodeName=" + node_name + ",env.nodeTaint=" + node_taint + ",env.nodeOsType=" + os_type])
+        values = "image.tag={},env.nodeName={},env.nodeTaint={},env.nodeOsType={}".format(image_tag, node_name, node_taint, os_type)  # pylint: disable=line-too-long
+
+        if service_principal is not None:
```
Comment (mboersma) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Would this be better as just 

```python
if not service_principal:
```

That's the pythonic way, unless we specifically want identity comparison with `None` here.

Comment (robbiezhang) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: do you mean

`if not service_principal:`

Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: D'oh yes of course that's what I meant.

**Pull request merged**